lib/retry.rb

```diff
@@ -17,4 +17,4 @@ def self.!(block, attempt=0)
 end
 
 class RetryableError < StandardError ; end
-class UnretryableError < StandardError ; end
\ No newline at end of file
+class UnretryableError < StandardError ; end
```

Comment:
Isn't this just a matter of discriminating on the retryable error? I.e.: The `UnretryableError' doesn't seem like it would be  necessary.

I suspect that there's specialized behavior for a RetryableError, but not for a StandardError. If that's the case, then the RetryableError specialization may be the only one that's needed. Unless there's specialized behavior for UnretryableError, it can be represented with a plain old StandardError.